src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ActionPlanService.kt

```diff
@@ -104,9 +104,9 @@ class ActionPlanService(
       ?: throw EntityNotFoundException("action plan not found [id=$id]")
   }
 
-  fun getActionPlanByReferral(referralId: UUID): ActionPlan {
-    return actionPlanRepository.findByReferralId(referralId)
-      ?: throw EntityNotFoundException("action plan not found [referralId=$referralId]")
+  fun getApprovedActionPlansByReferral(referralId: UUID): List<ActionPlan> {
+    return actionPlanRepository.findAllByReferralIdAndApprovedAtIsNotNull(referralId)
+      ?: throw EntityNotFoundException("no action plans found for referral [referralId=$referralId]")
```

Comment:
see other comment above - i would probably expect this to just return an empty list in this case